test/intl402/NumberFormat/prototype/format/value-decimal-string.js

```diff
@@ -18,7 +18,7 @@ assert.sameValue(nf.format('100000'), '100,000');
 // input is the string "-Infinity".
 assert.sameValue(nf.format('-100000'), '-100,000');
 
-assert.sameValue(nf.format('1.0000000000000001'), '1.0000000000000001');
-assert.sameValue(nf.format('-1.0000000000000001'), '1.0000000000000001');
+assert.sameValue(nf.format('1.0000000000000001'), '1');
+assert.sameValue(nf.format('-1.0000000000000001'), '-1');
```

Comment:
I think it should be

```
assert.sameValue(nf.format('1.0000000000000001'), '1.0000000000000001');
assert.sameValue(nf.format('-1.0000000000000001'), '-1.0000000000000001');
```

Reply:
Why is that? 
In SetNumberFormatDigitOptions
````
13.a Set hasSd to false.
....
16. ..... set needSd to false
...
22. Else,
22.a Set intlObj.[[RoundingType]] to morePrecision.
22.b Set intlObj.[[MinimumFractionDigits]] to 0.
22.c Set intlObj.[[MaximumFractionDigits]] to 0.
22.d Set intlObj.[[MinimumSignificantDigits]] to 1.
22.e Set intlObj.[[MaximumSignificantDigits]] to 2.
```

Reply:
You are correct.

The test is intended to test that string numbers work correctly, but it clearly isn't doing that.

Please fix this by adding `maximumFractionDigits: 21` to the constructor.

Reply:
> Please fix this by adding `maximumFractionDigits: 21` to the constructor.

Constructor will throw because
```
ii. Set mxfd to ? DefaultNumberOption(mxfd, 0, 20, undefined).
```
and 21 is larger than 20

Reply:
PTAL

Reply:
I can only add "{maximumFractionDigits: 20}" not "{maximumFractionDigits: 21}"